Settings.c

```diff
@@ -806,3 +806,25 @@ void Settings_setHeaderLayout(Settings* this, HeaderLayout hLayout) {
    this->hLayout = hLayout;
    this->changed = true;
 }
+
+static bool ProcessField_hasField(const ProcessField* fields, ProcessField field) {
+   if (fields) {
+      while (*fields) {
+         if (*fields++ == field) {
+            return true;
+         }
+      }
+   }
+   return false;
+}
+
+bool Settings_hasField(const Settings* this, ProcessField field) {
+   if (this->screens) {
+      for (unsigned int i = 0; this->screens[i]; i++) {
+         if (ProcessField_hasField(this->screens[i]->fields, field)) {
+            return true;
+         }
+      }
+   }
+   return false;
+}
```

Comment:
No need to look for information when the screen they are shown on is not visible either.

Reply:
Switched to using the accumulated flags as suggested and this function has been removed.